app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -23,12 +25,23 @@ import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
 
 enum class SearchStatuses {
-    SUCCESS, EMPTY_RESULT, CONNECTION_ERROR
+    SUCCESS, EMPTY_RESULT, CONNECTION_ERROR, IN_PROGRESS
 }
 
 const val PLAYLIST_HISTORY = "playlist_history"
 
 class SearchActivity : AppCompatActivity() {
+
+    private val CLICK_DEBOUNCE_DELAY = 1000L
+
+    private val SEARCH_DEBOUNCE_DELAY = 2000L
+
+    private val searchRunnable = Runnable { searchTrack() }
+
+    private var isClickAllowed = true
+
+    private val handler = Handler(Looper.getMainLooper())
```

Comment:
```suggestion
    private val handler by lazy { Handler(Looper.getMainLooper()) }
```
В качестве оптимизации можно воспользоваться `lazy`